apisix/plugins/limit-req.lua

```diff
@@ -67,7 +67,16 @@ function _M.access(conf, ctx)
         return 500
     end
 
-    local key = (ctx.var[conf.key] or "") .. ctx.conf_type .. ctx.conf_version
+    local key
+    if conf.key == "consumer_name" then
+        if not ctx.consumer_id then
+            core.log.error("The username of consumer is nil.")
+            return 401, { message = "Missing consumer's username."}
```

Comment:
the response code should be `500`. 

This problem occurs without authentication plug-ins. We can write the reason to the error log, it can help the user to confirm where is the problem.

Reply:
fixed